specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-11-01-preview/TuningOptions.json

```diff
@@ -0,0 +1,456 @@
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/flexibleServers/{serverName}/tuningOptions": {
+      "get": {
+        "tags": [
+          "TuningOptions"
+        ],
+        "operationId": "TuningOptions_ListByServer",
+        "x-ms-examples": {
+          "TuningOptions_ListByServer": {
+            "$ref": "./examples/Tuning_ListTuningOptions.json"
+          }
+        },
+        "description": "Retrieve the list of available tuning options.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../types/common-types.json#/parameters/ServerNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/TuningOptionsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ImpactRecord": {
+      "type": "object",
+      "properties": {
+        "dimensionName": {
+          "type": "string",
+          "description": "Dimension name"
+        },
+        "unit": {
+          "type": "string",
+          "description": "Dimension unit"
+        },
+        "queryId": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Optional property that can be used to store the QueryId if the metric is for a specific query."
+        },
+        "absoluteValue": {
+          "type": "number",
+          "format": "double",
+          "description": "Absolute value"
+        }
+      },
+      "description": "Stores property that features impact on some metric if this recommended action is applied."
+    },
+    "IndexRecommendationDetails": {
+      "type": "object",
+      "properties": {
+        "databaseName": {
+          "type": "string",
+          "description": "Database name."
+        },
+        "schema": {
+          "type": "string",
+          "description": "Schema name."
+        },
+        "table": {
+          "type": "string",
+          "description": "Table name."
+        },
+        "indexType": {
```

Comment:
>indexType [](http://example.com/codeflow?start=9&length=9)

Can this be modeled as enum with a fixed set of values?

Reply:
We do not have a fixed set of values for this property as of yet.

Reply:
If the set of values is fixed, but you don't yet know what they are, it may be worth prioritizing determining them before finalizing your swagger. Changing it in a future API version will require breaking change review.